### ISSUE TITLE
Major upgrade to 0.8.0 breaking react-native-context-menu builds on EAS. 

### ISSUE THREAD

**Issue by selfishfishi** [dominicstop/DGSwiftUtilities#1]
Hi @dominicstop 

We just noticed that our expo builds using react-native-context-menu is failing on EAS with the following error: 

```
Mon, 18 Dec 2023 00:20:58 GMT


❌  (ios/Pods/DGSwiftUtilities/Sources/Extension+EnumCaseStringRepresentable/UIUserInterfaceIdiom+EnumCaseStringRepresentable.swift:33:13)

  31 |         return "mac";
  32 | 
> 33 |       case .vision:
     |             ^ type 'UIUserInterfaceIdiom' has no member 'vision'
  34 |         return "vision";
  35 |       
  36 |       @unknown default:
  
  ````
  
  It seems that builds from a couple of hours ago with `0.7.1` were passing. 
  
  Is it possible that react-native-context-menu needs to be pinned to 0.7.1 ?

**Comment by dominicstop** [comment]
@selfishfishi hi, thank you for catching the issue (i'm currently trying to finish `react-native-ios-adaptive-modal` so i didn't notice this issue).

I committed this [fix](https://github.com/dominicstop/DGSwiftUtilities/commit/99e08116599f8d428452083a115ee5ff3b450358), and released a [patch (version 0.8.1)](https://github.com/dominicstop/DGSwiftUtilities/releases/tag/0.8.1) on cocoapods (it'll take some time before the new version is available).

Could you tell me what version of Xcode/Xcode CLI tools the expo EAS build environment is using? (i just assumed it's using [Xcode 15](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes)).

**Comment by selfishfishi** [comment]
Thanks for the quick response and quick path @dominicstop . I believe EAS is using `xcode-14.3.1` . I have attached the logs below. Thanks again ! 🙏  

```
Sun, 17 Dec 2023 20:18:33 GMT

Using VM template "macos-ventura-13.4-xcode-14.3.1" running macOS Ventura 13.4
Sun, 17 Dec 2023 20:18:33 GMT

Sun, 17 Dec 2023 20:18:33 GMT

Installed software:
Sun, 17 Dec 2023 20:18:33 GMT

- Xcode 14.3.1 (14E300c)
Sun, 17 Dec 2023 20:18:33 GMT

- Node.js 18.18.0
Sun, 17 Dec 2023 20:18:33 GMT

- Bun 1.0.14
Sun, 17 Dec 2023 20:18:33 GMT

- Yarn 1.22.19
Sun, 17 Dec 2023 20:18:33 GMT

- pnpm 8.6.5
Sun, 17 Dec 2023 20:18:33 GMT

- npm 9.8.1
Sun, 17 Dec 2023 20:18:33 GMT

- fastlane 2.213.0
Sun, 17 Dec 2023 20:18:33 GMT

- CocoaPods 1.12.1
Sun, 17 Dec 2023 20:18:33 GMT

- Ruby 2.7
Sun, 17 Dec 2023 20:18:33 GMT

- Python 3.12.0
Sun, 17 Dec 2023 20:18:33 GMT

- node-gyp 10.0.1
Sun, 17 Dec 2023 20:18:33 GMT

No `image` was explicitly specified. Resolved builder image to 'macos-ventura-13.4-xcode-14.3.1'.
Sun, 17 Dec 2023 20:18:33 GMT

- Expo CLI 6.3.10
```

**Comment by selfishfishi** [comment]
Hi @dominicstop 

This is all resolved now. Thank you very very much.
